### DISCUSSION TITLE
ChatGPT is bullshit

### DISCUSSION THREAD

**Discussion by sharethewisdom** [wincent/wincent#149] [General]
Hi Greg

I thought I'd share a recent philosophical paper that got some attention in the press. Given that you included a relevant contradiction in the ChatGPT prompt for [shellbot](https://github.com/wolffiex/shellbot).

You're prompting the LLM to assist without apologies or first-person  phrases, and you mention terminology that is used to describe attributes of consciousness. But then you prompt it with *ask yourself whether it might be a hallucination or not*, which not only contradicts the attributes you ask it to avoid, it also argues for the idea that these systems are misrepresenting the world [^1]
https://github.com/wincent/wincent/blob/c02f91d6821f687305fe4f03911d62c8fa36eeb6/aspects/dotfiles/files/.config/shellbot/prompt#L9

from the abstract of the cited paper:

> In this paper, we argue against the view that when ChatGPT and the like produce false claims they are lying or even hallucinating, and in favour of the position that the activity they are engaged in is bullshitting, in the Frankfurtian sense (Frankfurt, [2002](https://link.springer.com/article/10.1007/s10676-024-09775-5#ref-CR11), [2005](https://link.springer.com/article/10.1007/s10676-024-09775-5#ref-CR12)). Because these programs cannot themselves be concerned with truth, and because they are designed to produce text that looks truth-apt without any actual concern for truth, it seems appropriate to call their outputs bullshit. [^1]

[^1]: Hicks, M.T., Humphries, J. & Slater, J. ChatGPT is bullshit. Ethics Inf Technol 26, 38 (2024). [https://doi.org/10.1007/s10676-024-09775-5](https://doi.org/10.1007/s10676-024-09775-5)

**Comment by wincent** [comment]
Thanks for post, @sharethewisdom!

FWIW, I saw the "bullshit" paper when it [did the rounds](https://news.ycombinator.com/item?id=40626692) on [Hacker News](https://news.ycombinator.com/news) a while back, and I agree with you that "bullshit" is probably about right to describe what LLMs do when they produce untruths[^supplant].

For context, check out [the commit where I revised my prompt](https://github.com/wincent/wincent/commit/7e7115c89a80418895904a1c441ee8dcd52dcc85). I'm only telling it try not to hallucinate in a tongue in cheek way, because I don't think there actually _is_ a way to eliminate hallucinations from within — as argued by another paper to [recently surface](https://news.ycombinator.com/item?id=41541053) on HN, ["LLMs Will Always Hallucinate, and We Need to Live With This"](https://arxiv.org/html/2409.05746v1):

> This work argues that hallucinations in language models are not just occasional errors but an inevitable feature of these systems. We demonstrate that hallucinations stem from the fundamental mathematical and logical structure of LLMs. It is, therefore, impossible to eliminate them through architectural improvements, dataset enhancements, or fact-checking mechanisms. Our analysis draws on computational theory and Gödel’s First Incompleteness Theorem, which references the undecidability of problems like the Halting, Emptiness, and Acceptance Problems.

I get why you argue that there's a contradiction in my prompt, but I think you might be overlooking what the purpose of such a prompt is. It's not intended to be a robust and consistent philosophical argument; rather, it's intended to cue the machine to give me what I want. To that end, it doesn't matter whether the prompt is consistent, only that it produces the desired results.

In my case, I find it annoying when the LLM presents as if it were a conscious being — especially when it "begs for my forgiveness" and expresses "regret" for having made something up — so I tell it to avoid phrasing things in ways that frame things in terms of agency (etc), if possible. And at the end, I semi-ironically tell it not to hallucinate, not implying that I think it has any intentions of its own, nor even thinking that doing so will help, but perhaps to save me one round of having to ask it "is this X that you just told me about fictional, or real?" when it gives me a bullshit response[^really].

[^supplant]: Although it won't supplant the "hallucination" in common usage, because the latter is too well established right now, and can be used in contexts where vulgar words like "bullshit" cannot.
[^really]: Really, I don't think it will make any difference, and it's a far cry from the kinds ofexplorations of ["chain of thought" reasoning that OpenAI recently introduced](https://openai.com/index/learning-to-reason-with-llms/) (which look poised to deliver significant improvements in quality).

**Reply by sharethewisdom** [reply]
thanks, I didn't read article from the description in commit 7e7115c.

The irony is that the hallucination is not a property, but an attribute, thus, in the mind of the beholder. I don't know how a prompt about hallucination or uncertainty affects answer the current ChatGPT LLM.

---

Ok, I'm sorry, Greg, to bother you with this. Still, I'd like to further explain why I want to argue against the usage of anthropomorphic terms.

I studied musical instrument making, with a particular interest in restoration and preservation of historical remnants, and/or using these to make informed replicas.

Musical instruments are much more commonly attributed anthropomorphic properties, than other tools or items of clothing, not just in the anatomical naming of the components, like belly, neck, mouth, etc, ... but also related to non-use like resting, neglect, health, loosing its soul, etc. These rather emotional "artificial intelligence" attributes are possibly tied to emotions experienced in (classical) music performances.

The strong desire for expressing music often results in the irreversible destruction of historical remnants to make them current (according to fashion, a specific viewpoint) and playable, while in fact the best thing to do (to preserve them) is to not play them as much, and to take important measures like acclimatization and to closely monitor biological degradation. Luckily, a scientific approach from the field of archeology was becoming more common a century ago, and deciding what to do with an instrument in a certain condition often resulted in more informed restoration decisions that are reversible and well documented.[^1]

You're wondering now, why on earth is this relevant in a discussion about LLMs? In a way, I think our emotional desire for text is limited. But we probably feel strongly about the realism of stable diffusion outputs. I think that's where the desire to name it "hallucination" is strongest. But even when anthropomorphic attributes, such as reasoning and hallucinating are quantified[^2], people will continue to misunderstand that these LLMs are not equipped with a neurosymbolic deductive loop[^3] and that these are unaware of truth or reality, no matter the prompt.

At some point, probably when text-to-speach becomes a common LLM interface, people might subconsciously modify their behavior because they want to assist the LLM to "keep it healthy", to prevent it from "loosing its soul" and people might not notice that there is an irreversible amount of bullshit around on the internet. It will then be impossible to "restore" the internet to a previous state. People's voices might be used to improve conversational AI. Callcenters and radio stations might go out of business. More and more impressive bullshit would be generated until distinction between reality and representation vanishes.[^4]

We should therefore probably manage our programming expectations until there is a closed neurosymbolic loop or integrated tooling using linters or something.

[^1]: Barclay R.L. The Preservation and Use of Historic Musical Instruments : Display Case and Concert Hall. Earthscan; 2005.
[^2]: [A Multitask, Multilingual, Multimodal Evaluation of ChatGPT on Reasoning, Hallucination, and Interactivity](https://arxiv.org/abs/2302.04023) see also [github](https://github.com/HLTCHKUST/chatgpt-evaluation)
[^3]: [Are Hallucinations Popping the AI Bubble?](https://youtu.be/cHcWwGijXVc?feature=shared)
[^4]: [Simulacra and Simulation](https://en.wikipedia.org/wiki/Simulacra_and_Simulation)
